### PR TITLE
password hardening: Fix the test to support Debian 12

### PR DESCRIPTION
**Pull request by davidpil2002**
Fix the test to support Debian 12 by replacing cracklib usage with pwquality library

### Description of PR

Summary:
Fixes # (issue)
Fix as part of Debian 12 support in SONiC
### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [X ] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [ ] 202205
- [ ] 202305

#### What is the motivation for this PR?
Fix Password Hardening tests to support Debian 12 modifications.
#### How did you do it?
replace usage of pam_cracklib.so with pam_pwquality.so
#### How did you verify/test it?
run these PW hardening tests with the branch 112023 which based Debian 12
#### Any platform specific information?
no
#### Supported testbed topology if it's a new test case?
no

### CODE REVIEW

File: tests/passw_hardening/sample/passw_hardening_digits/common-password [hunk 1]
```diff
@@ -24,7 +24,7 @@
 
 # here are the per-package modules (the "Primary" block)
 
-password    requisite      pam_cracklib.so retry=3 maxrepeat=0 minlen=1 ucredit=0 lcredit=0 dcredit=-1 ocredit=0  enforce_for_root
+password    requisite      pam_pwquality.so retry=3 maxrepeat=0 minlen=1 ucredit=0 lcredit=0 dcredit=-1 ocredit=0  enforce_for_root dictcheck=0
```
[the remark this answers is not in the document]
Comment (dgsudharsan) [reply tests/passw_hardening/sample/passw_hardening_digits/common-password:27]: Hi @saiarcot895  We have a separate branch for 202305. Hence I don't think this is required https://github.com/sonic-net/sonic-mgmt/tree/202305
The only thing is this test needs to be merged after bookworm changes are merged to master

[the remark this answers is not in the document]
Comment (saiarcot895) [reply tests/passw_hardening/sample/passw_hardening_digits/common-password:27]: Yes, that should be fine.

File: tests/passw_hardening/conftest.py [hunk 2]
```diff
@@ -71,9 +71,11 @@ def clean_passw_en_dis_policies(duthosts, enum_rand_one_per_hwsku_hostname):
     duthost = duthosts[enum_rand_one_per_hwsku_hostname]
     passw_hardening_utils.config_user(duthost=duthost, username=passw_hardening_utils.USERNAME_SIMPLE_0, mode='del')
     passw_hardening_utils.config_user(duthost=duthost, username=passw_hardening_utils.USERNAME_SIMPLE_1, mode='del')
+    passw_hardening_utils.config_user(duthost=duthost, username=passw_hardening_utils.USERNAME_SIMPLE_2, mode='del')
     passw_hardening_utils.config_user(duthost=duthost, username=passw_hardening_utils.USERNAME_STRONG, mode='del')
     duthost.shell('sed -i /^'+passw_hardening_utils.USERNAME_SIMPLE_0+':/d /etc/security/opasswd')
     duthost.shell('sed -i /^'+passw_hardening_utils.USERNAME_SIMPLE_1+':/d /etc/security/opasswd')
+    duthost.shell('sed -i /^'+passw_hardening_utils.USERNAME_SIMPLE_2+':/d /etc/security/opasswd')
     duthost.shell('sed -i /^'+passw_hardening_utils.USERNAME_STRONG+':/d /etc/security/opasswd')
```
Comment (saiarcot895) [inline tests/passw_hardening/conftest.py:79]: Is `/etc/security/opasswd` actually used by `pam_pwquality`? It looks like this file is used by `pam_history`, and `pam_pwquality` can only check the previous password if it's entered on the command line, based on [this](https://github.com/libpwquality/libpwquality/issues/57).

Comment (davidpil2002) [reply tests/passw_hardening/conftest.py:79]: you test the functionaly of password hardening, the create new users with different passwords,
once you create a user in Linux, this password may be save in this file to be used later for pam_history.
The required to clean every modification in the system, so its cleaning this file as well.

**Review by davidpil2002 — COMMENTED** [review]
answered all comments

**Comment by dgsudharsan** [comment]
@saiarcot895  Can you please review and signoff?

**Pull request merged**